plume/src/main/kotlin/io/github/plume/oss/passes/graph/BaseCPGPass.kt

```diff
@@ -660,6 +711,54 @@ class BaseCPGPass(private val g: BriefUnitGraph) {
         }
     }
 
+    private fun projectFieldAccess(fieldRef: FieldRef, childIdx: Int): NewCallBuilder{
+        val fieldAccessVars = mutableListOf<NewNodeBuilder>()
+        val leftOp = when(fieldRef){
+            is StaticFieldRef ->  fieldRef.fieldRef.declaringClass().type
+            is InstanceFieldRef -> fieldRef.base.type
+            else -> fieldRef.fieldRef.declaringClass().type
+        }
+
+        val fieldAccessBlock = NewCallBuilder()
+                .name(Operators.fieldAccess)
+                .code(fieldRef.toString()) // Review-TODO: Check Code is valid
+                .signature("(${leftOp.toQuotedString()}).(${fieldRef.toString()})") // Review-TODO: Signature of  [(Static/Instance) Field = Signature of Right Tree]
+                .methodFullName(Operators.fieldAccess)
+                .dispatchType(if(fieldRef is StaticFieldRef) DispatchTypes.STATIC_DISPATCH else DispatchTypes.DYNAMIC_DISPATCH)   // Review-TODO: How to handle which instance(Variable) is being used?
+                .dynamicTypeHintFullName(SootToPlumeUtil.createScalaList(leftOp.toQuotedString()))
+                .order(childIdx)
+                .argumentIndex(childIdx)
+                .typeFullName(leftOp.toQuotedString())
+                .lineNumber(Option.apply(currentLine))
+                .columnNumber(Option.apply(currentCol))
+                .apply{ fieldAccessVars.add(this) }
+        when(fieldRef){
+            is StaticFieldRef ->{ // Handle Static as Type_ref?
+                Pair(SootToPlumeUtil.createTypeRefVertex(fieldRef.type, currentLine, currentCol, 0),  // Type_Ref childIdx 1? cuz 0 is reserved.. for this as schema document
+                        SootToPlumeUtil.createFieldIdentifierVertex(fieldRef,  currentLine, currentCol, 1))  // FieldIdentifier, chilIdx 2?
+            }
+            is InstanceFieldRef -> { // Handle Local? and Identifier?
+                Pair(SootToPlumeUtil.createIdentifierVertex(fieldRef.base, currentLine, currentCol, 0),
+                        SootToPlumeUtil.createFieldIdentifierVertex(fieldRef, currentLine, currentCol, 1))
+            }
+            else -> null
+        }?.let {
+            it.toList().asSequence().forEach {
+                runCatching {  //Review-TODO: Need more Edges? or  can we pass them to scpg-passes
+                    builder.addEdge(fieldAccessBlock, it, AST)
+                    builder.addEdge(fieldAccessBlock, it, ARGUMENT)
+                    fieldAccessVars.add(it)
+                }.onFailure { e -> logger.warn(e.message) }
+            }
+        }
+
+        // Call for <op>.fieldAccess, cast doesn't need <RECEIVER>?
```

Comment:
I think it may need a `<RECEIVER>`. Around line 561 where the edges are added, you can check if it's a field assignment by doing
```kotlin
// Existing code
builder.addEdge(assignBlock, it, AST)
builder.addEdge(assignBlock, it, ARGUMENT)
// New code
val operator = it.build().properties()[NAME]
if (operator.isDefined && operator.get() == Operators.fieldAccess) builder.addEdge(assignBlock, it, RECEIVER)
```

Reply:
Handeld them as following
if(it is NewCallBuilder) builder.addEdge(assignBlock, it, RECEIVER)